File: starters/JavaScript/MyBot.js
const process = require('process');
const bomberjam = require('./bomberjam');

const allActions = [
  bomberjam.Constants.Left,
  bomberjam.Constants.Up,
  bomberjam.Constants.Right,
  bomberjam.Constants.Down,
  bomberjam.Constants.Stay,
  bomberjam.Constants.Bomb
];

const game = new bomberjam.Game();

// Standard output (console.log) can ONLY BE USED to communicate with the bomberjam process
// Use text files if you need to log something for debugging
const logger = new bomberjam.Logger();

// Edit run_game.(bat|sh) to include file logging for any of the four bot processes: node MyBot.js --logging
if (process.argv.slice(2).some(x => x.toLowerCase() === '--logging'))
  logger.setup(`log-${new Date().getTime()}.log`);

const main = async () => {
  // 1) You must send an alphanumerical name up to 32 characters
  // Spaces or special characters are not allowed
  await game.ready('MyName' + Math.round(Math.random() * (9999 - 1000) + 1000));
  logger.info('My player ID is ' + game.myPlayerId);

  do {
    // 2) Each tick, you'll receive the current game state serialized as JSON
    // From this moment, you have a limited time to send an action back to the bomberjam process through stdout
    await game.receiveCurrentState();

    try {
      // 3) Analyze the current state and decide what to do
      const state = game.state;

      for (let x = 0; x < state.width; x++) {
        for (let y = 0; y < state.height; y++) {
          const tile = bomberjam.StateUtils.getTileAt(state, x, y);
          if (tile === bomberjam.Constants.Block) {
            // TODO found a block to destroy
          }

          const otherPlayer = bomberjam.StateUtils.findAlivePlayerAt(state, x, y);
          if (otherPlayer && otherPlayer.id !== game.myPlayerId) {
            // TODO found an alive opponent
          }

          const bomb = bomberjam.StateUtils.findActiveBombAt(state, x, y);
          if (bomb) {
            // TODO found an active bomb
          }

          const bonus = bomberjam.StateUtils.findDroppedBonusAt(state, x, y);
          if (bonus) {
            // TODO found a bonus
          }
        }
      }

      if (game.myPlayer.bombsLeft > 0) {
        // TODO you can drop a bomb
      }

      // 4) Send your action
      const action = allActions[Math.floor(Math.random() * allActions.length)];
      game.sendAction(action);
      logger.info('Tick ' + state.tick + ', sent action: ' + action);
    } catch (err) {
      // Handle your exceptions per tick
      logger.error('Tick ' + game.state.tick + ', exception: ' + err.toString());
    }
  } while (game.myPlayer.isAlive && !game.state.isFinished);
};

main().catch(console.err).finally(() => {
  logger.close();
  game.close();
});

File: starters/Python3/MyBot.py
import random
import sys
import time

from bomberjam.constants import Constants
from bomberjam.game import Game
from bomberjam.logger import Logger


game = Game()

# Standard output (print) can ONLY BE USED to communicate with the bomberjam process
# Use text files if you need to log something for debugging
logger = Logger()

# Edit run_game.(bat|sh) to include file logging for any of the four bot processes: python3 MyBot.py --logging
if True in (arg.lower() == '--logging' for arg in sys.argv):
    logger.setup('log-' + str(round(time.time() * 1000)) + '.log')

# 1) You must send an alphanumerical name up to 32 characters
# Spaces or special characters are not allowed
game.ready("MyName" + str(random.randint(1000, 9999)))
logger.info('My player ID is ' + game.my_player_id)

while True:
    # 2) Each tick, you'll receive the current game state serialized as JSON
    # From this moment, you have a limited time to send an action back to the bomberjam process through stdout
    game.receive_current_state()

    try:
        # 3) Analyze the current state and decide what to do
        for x in range(game.state.width):
            for y in range(game.state.height):
                tile = game.state.get_tile_at(x, y)
                if tile == Constants.BLOCK:
                    pass  # TODO found a block to destroy

                other_player = game.state.find_alive_player_at(x, y)
                if other_player is not None and other_player.id != game.my_player_id:
                    pass  # TODO found an alive opponent

                bomb = game.state.find_active_bomb_at(x, y)
                if bomb is not None:
                    pass  # TODO found an active bomb

                bonus = game.state.find_dropped_bonus_at(x, y)
                if bonus is not None:
                    pass  # TODO found a bonus

        if game.my_player.bombs_left > 0:
            pass  # TODO you can drop a bomb

        # 4) Send your action
        action = random.choice([Constants.UP, Constants.DOWN, Constants.LEFT, Constants.RIGHT, Constants.STAY, Constants.BOMB])
        game.send_action(action)
        logger.info('Tick ' + str(game.state.tick) + ', sent action: ' + action)
    except Exception as ex:
        # Handle your exceptions per tick
        logger.error('Tick ' + str(game.state.tick) + ', exception: ' + str(ex))

    if not game.my_player.is_alive or game.state.is_finished:
        break
